numpy/core/src/multiarray/item_selection.c

```diff
@@ -1749,6 +1750,275 @@ PyArray_LexSort(PyObject *sort_keys, int axis)
     return NULL;
 }
 
+/*NUMPY_API
+ *Keysort an array providing indices that will sort a collection of arrays
+ *lexicographically.  The first key is sorted on first, followed by the second key
+ *-- requires that arg"merge"sort is available for each sort_key
+ *
+ *Returns None on success, returns NULL on failure
+ */
+NPY_NO_EXPORT PyObject *
+PyArray_KeySort(PyArrayObject *self, PyObject *sort_keys, int axis)
+{
+    PyArrayObject **mps;
+    PyArrayIterObject **its;
+    PyArrayIterObject *sit = NULL;
+    npy_intp n, N, size, i, j;
+    npy_intp astride, *iptr, selstride;
+    int nd;
+    int needcopy = 0;
+    int elsize;
+    int aelsize;
+    int maxelsize;
+    int object = 0;
```

Comment:
Can we push these declarations down-file to where they are first used, C99-style?

Reply:
Not sure on this one, every other function in the same file does not follow it, shouldn't be a problem changing, only question is uniformity.

Reply:
That's because other functions were written before the language allowed it. IMO we should write all new code in c99 style.